score-powershell/src/main/java/io/cloudslang/content/actions/PowerShellScriptAction.java

```diff
@@ -0,0 +1,136 @@
+            }
+    )
+    public Map<String, String> execute(
+            @Param(value = INPUT_HOST, required = true) String host,
+            @Param(value = INPUT_PORT) String port,
+            @Param(value = PROTOCOL) String protocol,
+            @Param(value = USERNAME, required = true) String username,
+            @Param(value = PASSWORD, required = true, encrypted = true) String password,
+            @Param(value = PROXY_HOST) String proxyHost,
+            @Param(value = PROXY_PORT) String proxyPort,
+            @Param(value = PROXY_USERNAME) String proxyUsername,
+            @Param(value = PROXY_PASSWORD, encrypted = true) String proxyPassword,
+            @Param(value = TRUST_ALL_ROOTS) String trustAllRoots,
+            @Param(value = X509_HOSTNAME_VERIFIER) String x509HostnameVerifier,
+            @Param(value = TRUST_KEYSTORE) String trustKeystore,
+            @Param(value = TRUST_PASSWORD, encrypted = true) String trustPassword,
+            @Param(value = KEYSTORE) String keystore,
+            @Param(value = KEYSTORE_PASSWORD, encrypted = true) String keystorePassword,
+            @Param(value = MAX_ENVELOP_SIZE) String maxEnvelopeSize,
+            @Param(value = INPUT_SCRIPT, required = true) String script,
+            @Param(value = WINRM_LOCALE) String winrmLocale,
+            @Param(value = OPERATION_TIMEOUT) String operationTimeout
+    ) {
+        Map<String, String> resultMap = new HashMap<>();
+        try {
+            WSManRemoteShellService wsManRemoteShellService = new WSManRemoteShellService();
+            protocol = StringUtils.isBlank(protocol) ? InputDefaults.PROTOCOL.getValue() : protocol;
+            port = StringUtils.isBlank(port) ? InputDefaults.PORT.getValue() : port;
+            maxEnvelopeSize = StringUtils.isBlank(maxEnvelopeSize) ? InputDefaults.MAX_ENVELOPE_SIZE.getValue() : maxEnvelopeSize;
+            Integer.parseInt(maxEnvelopeSize);
+            trustAllRoots = StringUtils.isBlank(trustAllRoots) ? InputDefaults.TRUST_ALL_ROOTS.getValue() : trustAllRoots;
+            x509HostnameVerifier = StringUtils.isBlank(x509HostnameVerifier) ? InputDefaults.X_509_HOSTNAME_VERIFIER.getValue() : x509HostnameVerifier;
+            operationTimeout = StringUtils.isBlank(operationTimeout) ? InputDefaults.OPERATION_TIMEOUT.getValue() : operationTimeout;
+            winrmLocale = StringUtils.isBlank(winrmLocale) ? InputDefaults.WINRM_LOCALE.getValue() : winrmLocale;
+
+            resultMap = wsManRemoteShellService.runCommand(host, port, protocol, username, password, proxyHost,
+                    proxyPort, proxyUsername, proxyPassword, maxEnvelopeSize, trustAllRoots, x509HostnameVerifier, keystore, keystorePassword,
+                    trustKeystore, trustPassword, script, winrmLocale, Integer.parseInt(operationTimeout));
```

Comment:
maybe you could use an object to hold all these inputs, see score-vmware on how the builder pattern is used there for this purpose